nova_vm/src/heap/heap_gc.rs

```diff
@@ -1008,6 +1012,10 @@ fn sweep(agent: &mut Agent, bits: &HeapBits, root_realms: &mut [Option<RealmIden
         builtin_functions,
         #[cfg(feature = "array-buffer")]
         data_views,
+        #[cfg(feature = "array-buffer")]
+            data_view_byte_lengths: _,
```

Comment:
**issue:** So here we cannot ignore the byte length and byte offsets.

Above in the heap_gc call we do not need to trace through these, as there are never any references in the byte lengths and offsets.

But here when we sweep and compact the heap, here we do need to do the index shifting work and removing dropped indexes.